src/vcpkg/tools.cpp

```diff
@@ -25,25 +25,57 @@ namespace vcpkg
         std::string sha512;
     };
 
-    static Optional<std::array<int, 3>> parse_version_string(const std::string& version_as_string)
+    // /\d+\.\d+(\.\d+)?/
+    static Optional<std::array<int, 3>> parse_version_string(StringView string_version)
     {
-        static const std::regex RE(R"###((\d+)\.(\d+)(\.(\d+))?)###");
+        using P = Parse::ParserBase;
 
-        std::match_results<std::string::const_iterator> match;
-        const auto found = std::regex_search(version_as_string, match, RE);
-        if (!found)
+        // first, find the beginning of the version
+        auto first = string_version.begin();
+        auto last = string_version.end();
+
+        // we're looking for the first instance of `<digits>.<digits>`
```

Comment:
Should we fix `try_parse_external_dot_version` to allow this to work without this search instead?

Reply:
no, because `try_parse_external_dot_version` also needs to work with only a single version number (it needs to parse `3` as a dotted version.